File: examples/lua53/client/protocols/ReqSceneEnter.lua
local M = {}
local moduleName = ...
_G[moduleName] = M
setmetatable(M, {__index = _G})
local _ENV = M


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.door_id = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.door_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER)
end


function setDoorId(self, door_id)
	self.door_id = door_id
end

File: examples/CocosCreatorTs/assets/scripts/proto/ReqSceneEnter.ts
import { Packet } from '@mi/mod/Packet'


export class ReqSceneEnter
{
	private _door_id: number;


	public Encode(): Packet {
		let packet: Packet = new Packet();
		packet.WriteUint(this._door_id);
		packet.Encode(2020);
		return packet;
	}


	public get door_id(): number { return this._door_id; }
	public set door_id(value: number) { this._door_id = value; }
}
